pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeSpawner.java

```diff
@@ -98,19 +107,19 @@ public void join() throws Exception {
         }
     }
 
-    public CompletableFuture<FunctionStatus> getFunctionStatus() {
-        return runtime.getFunctionStatus().thenApply(f -> {
-            FunctionStatus.Builder builder = FunctionStatus.newBuilder();
-            builder.mergeFrom(f).setNumRestarts(numRestarts).setInstanceId(instanceConfig.getInstanceName());
+    public CompletableFuture<FunctionStatus> getFunctionStatus(int instanceId) {
```

Comment:
Why do we need to pass the instanceId.  The instanceConfig should have that information

Reply:
Reason is that for an externally managed runtime like kubernetes, the runtime spawner is actually launching an external job which contains the actual instances. Thus we need to pass the instance id of the function of whose data we want